chapters/01-preface.Rmd

```diff
@@ -0,0 +1,334 @@
+of their own. Throughout the book there are sections dedicated to this reader
+that give information or instructions on how to use or modify the material for
+the instructor's specific context.
+
+## Introduction for learners
+
+This primary aim of this book is to teach you the tools, skills, and knowledge
+necessary to making research software that is open, reproducible, and
+sustainable. For researchers and data scientists who can write functions to
+create programs that are hundreds of lines long and who want to be more
+productive and have more confidence in their results, this book provides a
+pragmatic, tools-based introduction to research software engineering. 
+For researchers whose tasks include building software packages, this book will
+help train you to be a [research software engineer][rse]. Unlike courses aimed
+at computer scientists and professional software developers, this book uses data
+analysis as the motivating example and assumes that the learner's ultimate goal
+is to answer questions rather than ship products.
+
+First and foremost, the main learning goal of this course is that you learn to
+*develop an R package*. More specifically, the **learning objectives** are to:
+
+- Explain what an R package is, how it is structured, and how to set one up.
+- List and describe some best (or good enough) practices in package
+development and how to implement them. 
+- Explain what a code style guide is, why it is important, and how to write code
+that adheres to the style guide.
+- Describe why user-friendly and well-written documentation is crucial to
+building a usable package and some steps to achieve this type of
+documentation.
+- Cover what CRAN is, what policies and quality-control steps it has for packages,
+and how to prepare a package for submission to CRAN so that it passes these checks
+and requirements.
+
+Aside from the technical aspects of building an R package, the course
+will also cover topics that enhance and improve the process of package development.
+These learning objectives are to:
+
+- Describe and apply effective workflows and project management tools that
+coordinate the development of software, particularly for software that is
+complex or when working in a team.
+- Outline (and optionally practice) how to work productively in a small team
+that is inclusive and welcoming.
+- Learning why and how to design software and documentation
+involving community contributors.
+
+By the end of the course, you will have the skills and knowledge necessary to:
+build your own packages that are well-built, well-documented, and re-usable;
+work effectively as part of a team or as a contributor; and, if relevant, prepare
+your package to be submitted to CRAN. 
+
+TODO: Add graphic for overview? Something like https://moderndive.com/preface.html#fig:pipeline-figure?
+
+### Who is this course for? {#learner-personas}
+
+The following personas are examples of the types of people that are our target audience.
```

Comment:
Given we are framing the book for use in an undergraduate or graduate class, I'm wondering if we should adjust one of the learner profiles to represent a student specifically taking a class, and including their intended career objectives?